### ISSUE TITLE
The decimal format (d) isn’t observed if x ≥ 1e21.

### ISSUE THREAD

**Issue by mbostock** [d3/d3-format#24]
For example:

``` js
var f = d3.format(",d");
f(1e20); // "100,000,000,000,000,000,000"
f(1e21); // "1e+21"
```

Because:

``` js
(1e20).toString(10); // "100000000000000000000"
(1e21).toString(10); // "1e+21"
```

Related #21.

**Comment by mbostock** [comment]
On the other hand, toLocaleString seems to do what we want:

```js
(1e21).toLocaleString("en") // "1,000,000,000,000,000,000,000"
```

**Comment by mbostock** [comment]
Fixed in #104.
